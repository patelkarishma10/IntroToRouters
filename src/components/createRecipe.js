import React, { Component } from 'react';
import axios from "axios";

export class CreateRecipe extends Component {
createRecipe = (e) => {
            e.preventDefault();
     let newItem = {
            name: e.target[0].value,
            description: e.target[1].value,
            ingredients: e.target[2].value,
            image: e.target[3].value
        }

        axios
            .post("http://localhost:5000/recipe/createRecipe", newItem)

            .then(response => {
                console.log("created recipe")

                this.props.passedFunction();
            });

}

    render() {

        return (
            <div className="formDiv">
                <h1>Create Recipe</h1>
                <form onSubmit={this.createRecipe}>
                    <label htmlFor="name">Name: </label>
                    <input type="text" id="name" className="form-control" required />
                    <br />

                    <label htmlFor="description" >Description: </label>
                    <input type="content" id="description" className="form-control" required />
                    <br />

                    <label htmlFor="ingredients" >Ingredients: </label>
                    <input type="content" id="ingredients" className="form-control" required />
                    <br />

                    <label htmlFor="image" >Image: </label>
                    <input type="content" id="image" className="form-control" required />
                    <br />

                    <input type="submit" className="btn btn-dark" value="create recipe" />
                </form>
            </div>

        );
    }
}