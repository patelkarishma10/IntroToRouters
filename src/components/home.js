import React, { Component } from 'react';


export class Home extends Component {

  render() {
    return (
      <div>
       <p>dfgd</p>
        
      </div>
    );
  }


}

export default Home;