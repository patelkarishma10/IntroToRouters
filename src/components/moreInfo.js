import React, { Component } from 'react';
import { Table } from "reactstrap";

export class MoreInfo extends Component {

    render() {
        return (
            <div class="formDiv">
                <p>Recipe: {this.props.name}</p>
                <p>Description: {this.props.description}</p>
                <p>Ingredients: {this.props.ingredients}</p>
            </div>

        );
    }
}