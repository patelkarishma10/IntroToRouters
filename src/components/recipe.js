import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
//import {MoreInfo} from './moreInfo';
// import Home from './home'


export class Recipe extends Component {
    render(props) {

const { 
    name 
} = this.props

        return (
                    <tbody>
                        <tr>
                            <td>
                                 <button  className="btn btn-dark">
                            <Link to={name}>{name}</Link>
                            </button>
                            </td>
                        </tr>

                    </tbody>

        );
    }
}
