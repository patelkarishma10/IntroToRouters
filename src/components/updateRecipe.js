import React, { Component } from 'react';
import axios from "axios";

export class UpdateRecipe extends Component {
UpdateRecipe = (e) => {
            e.preventDefault();
     let newItem = {
            name: e.target[0].value,
            description: e.target[1].value,
            ingredients: e.target[2].value,
            image: e.target[3].value,
            _id: this.props._id
        }

        axios
            .put("http://localhost:5000/recipe/updateRecipe", newItem)

            .then(response => {
                console.log("created recipe")

                this.props.passedFunction();
            });

    }

    render() {

        return (
            <div className="formDiv">
                <h2>Update Recipe</h2>
                <form onSubmit={this.updateRecipe}>
                    <label htmlFor="name">Name: </label>
                    <input type="text" id="name" className="form-control" required />
                    <br />

                    <label htmlFor="description" >Description: </label>
                    <input type="content" id="description" className="form-control" required />
                    <br />

                    <label htmlFor="ingredients" >Ingredients: </label>
                    <input type="content" id="ingredients" className="form-control" required />
                    <br />
                    <label htmlFor="image" >Image: </label>
                    <input type="content" id="image" className="form-control" required />
                    <br />
                    <input type="submit" className="btn btn-dark" value="create recipe" />
                </form>
            </div>

        );
    }
}