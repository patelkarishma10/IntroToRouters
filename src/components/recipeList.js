import React, { Component } from 'react';
import {Table} from "reactstrap";
import { Recipe} from './recipe';
// import { BrowserRouter as Router, Route, Link } from "react-router-dom";


export class RecipeList extends Component {

    render(props) {
        return (
            <div className="formDiv">
                <div id="random"> </div>
                <h2>Recipes</h2>
                <Table>
                        {this.props.dataSent.map((item, index) => (

                                 <Recipe key={index} passedFunction={this.props.passedFunction} name={item.name} description={item.description} ingredients={item.ingredients} image={item.image}/>       

                        ))}
                </Table>
            </div>

        );
    }
}

