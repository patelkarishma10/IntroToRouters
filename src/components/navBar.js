import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import axios from "axios";

import { CreateRecipe } from './createRecipe';
import { RecipeList } from './recipeList';
import { MoreInfo } from './moreInfo';
// import { Home } from './home';

export class NavBar extends Component {

      constructor() {
    super();
    this.state = {
      data: []
    };
  }
  

  onLoad = () => {
        axios
      .get("http://localhost:5000/recipe/all")
      .then(response => {
        this.setState({
          data: response.data
          
        });
      })
 
      }

      componentDidMount() {
        this.onLoad();

      }

      render() {
        return (
          <div>
            <Router>
              <ul className="navBar">
                <li id="homeButton" ><Link to="/">Home</Link></li>
                <li id="createButton"><Link to="/create">Create Recipes</Link></li>
              </ul>
              <Route path="/create" render={(props) => <CreateRecipe passedFunction={this.onLoad} dataSent={this.state.data} />} />
              <Route exact path="/" render={(props) => <RecipeList passedFunction={this.onLoad} dataSent={this.state.data} />} />


              {this.state.data.map((item, index) => (

                <Route key={index} path={"/"+item.name} render={(props) => <MoreInfo passedFunction={this.onLoad}
                  description={item.description}
                  ingredients={item.ingredients}
                  name={item.name}
                  image={item.image}
                  id={item._id} />} />

              ))}
            </Router>
          </div>
        )
      }
}
