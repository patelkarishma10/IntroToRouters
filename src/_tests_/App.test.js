import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from '../App';
import { NavBar } from '../components/navBar';

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders NavBar component', () => {
  const wrapper = shallow(<App />);
  const content = <NavBar />;
  expect(wrapper.contains(content)).toEqual(true);
});

it('snapshot App', () => {
  // const wrapper = shallow(<App />);
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot();
})

