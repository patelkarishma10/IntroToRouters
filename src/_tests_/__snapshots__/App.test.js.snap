// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App 1`] = `
<div>
  <div>
    <ul
      className="navBar"
    >
      <li
        id="homeButton"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        id="createButton"
      >
        <a
          href="/create"
          onClick={[Function]}
        >
          Create Recipes
        </a>
      </li>
    </ul>
    <div
      className="formDiv"
    >
      <div
        id="random"
      >
         
      </div>
      <h2>
        Recipes
      </h2>
      <table
        className="table"
      />
    </div>
  </div>
</div>
`;
